api/v1alpha4/azurecluster_default.go

```diff
@@ -148,6 +151,58 @@ func (c *AzureCluster) setAPIServerLBDefaults() {
 	}
 }
 
+func (c *AzureCluster) setNodeOutboundLBDefaults() {
+	if c.Spec.NetworkSpec.NodeOutboundLB == nil {
+		if c.Spec.NetworkSpec.APIServerLB.Type == Internal {
+			return
+		}
+		c.Spec.NetworkSpec.NodeOutboundLB = &LoadBalancerSpec{}
+	}
+
+	lb := c.Spec.NetworkSpec.NodeOutboundLB
+	if lb.Type == "" {
+		lb.Type = Public
+	}
+	if lb.SKU == "" {
+		lb.SKU = SKUStandard
+	}
+
+	if lb.Name == "" {
+		lb.Name = c.ObjectMeta.Name
```

Comment:
there's a catch here... Azure only allows you to attach one public (and one private) LB to each VM. Because the cloud provider already creates a public LB (and internal one) for worker nodes to expose services of type LoadBalancer, we have to use the same LB and can't create a separate one. Cloud provider makes the assumption that the LB name is <CLUSTER_NAME> https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/site/content/en/topics/assumptions.md so this isn't technically configurable and has to be `<CLUSTER_NAME>` (or whatever `--cluster-name` is set to in cloud-controller-manager https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/templates/flavors/base/cluster-template.yaml#L73).

Reply:
Thanks for the explanation, Cecile. I was wondering why we didn't have any suffix to node outbound lb name like we do for api-server. This makes sense. Will fix it. Probably will end up disabling mutation as suggested in your other comment.